docs/example_csharp.md

```diff
@@ -1,10 +1,256 @@
+      Now listening on: http://[::]:
+```
+
+{% assign case = "base" %}
+{% include example_guide_image.html %}
+
+## Step 1: Let's Add Benchmark Trickery
+
+Before we try to make this faster, let's measure it.
+
+Using [`local_resource`](local_house.html), you can direct Tilt to execute existing scripts or arbitrary shell commands on your own machine, and manage them from your sidebar like any other Tilt resource. We're going to use this functionality to benchmark our deploys.
+
+First we add a `local_resource` to our [Tiltfile](https://github.com/windmilleng/tilt-example-csharp/blob/master/1-measure/Tiltfile) that records the start time in a C# file.
+
+```python
+local_resource(
+    'deploy',
+    './record-start-time.sh',
+    deps=['./record-start-time.sh']
+)
+```
+
+The `local_resource()` call creates a local resource named `deploy`. The second argument is the script that it runs.
+
+We've also modified our server to read that start itme, calculate the time elapsed, then display this in the rendered HTML.
+
+Let's click the button on the `deploy` resource and see what happens!
+
+<figure>
+  <a class="is-image" href="https://cloud.tilt.dev/snapshot/AdKa5ucLVfFH0lRxn7I=">
+    <img src="assets/docimg/example-csharp-image-2.png">
+  </a>
+  <figcaption>Step 1 complete. Click the screenshot to see an interactive snapshot.</figcaption>
+</figure>
+
+| Approach | Deploy Time[^1] |
+|---|---|
+| Naive | 10.4s |
+{:.benchmark-report}
+
+If you look closely, the elapsed time displayed in the Tilt sidebar is different than the benchmark our app logged. That's OK! In multi-service development there are many benchmarks we care about -- the time to build the image, the time to schedule the process, and the time until the server is ready to serve traffic.
+
+Tilt offers you some default benchmarks _and_ the tools to capture your own.
+
+Our benchmarks show this is slow. Can we do better?
+
+## Step 2: Let's Optimize for the C# Toolchain
+
+What's taking up so much time? The logs show that when we make a change to a file, we:
+
+1) Copy the csproj file, run `dotnet restore` to install any dependencies
+2) Copy the rest of the code and run a build from scratch with `dotnet publish`
+3) Copy the build output to the ASP.NET runtime image
+
+But the C# community has done a lot of work to make caching dependendencies and incremental compiles fast. How can we better use the tools how they're meant to be used?
+
+With `local_resource`, we can compile the project locally, and copy the build output files to the container.
+
+Here's our [new Tiltfile](https://github.com/windmilleng/tilt-example-csharp/blob/master/2-build-locally/Tiltfile) with the following new code:
+
+```python
+local_resource(
+    'build',
+    'dotnet publish -c Release -o out',
+    deps=['hello-tilt'],
+    ignore=['hello-tilt/obj'],
+    resource_deps=['deploy'],
+)
+```
+
+We've added a `local_resource()` that compiles the executable Jar locally with Gradle.
```

Comment:
did you need to adjust the build context for your `docker_build` at all? might be useful to mention--otherwise users will see duplicative builds and not necessarily know how to debug.

Reply:
Yes! Good call, I'll add a note about that.